hi-ml/src/health_ml/utils/logging.py

```diff
@@ -333,6 +317,107 @@ def _print(self, message: str) -> None:
             sys.stdout.flush()
 
 
+class MLFlowLogger(LightningLoggerBase):
+    HYPERPARAMS_NAME = "hyperparams"
+
+    def _start_run(self, mlflow_run_id: Optional[str] = None) -> None:
+        """
+        Start an MLFlow run. If an existing run id is given,
+
+        :param mlflow_run_id: _description_, defaults to None
+        :return: _description_
+        """
+        mlflow.set_tracking_uri(self.mlflow_uri)
+        mlflow.set_experiment(self.experiment_name)
+        run = mlflow.start_run(run_id=mlflow_run_id)
+        return run
+
+    def __init__(self, experiment_name: Optional[str] = None, mlflow_uri: Optional[str] = None,
+                 run: Optional[str] = None):
+        experiment_name = experiment_name or os.environ.get("MLFLOW_EXPERIMENT_NAME")
+        self.experiment_name = experiment_name
+        mlflow_uri = mlflow_uri or os.environ.get("MLFLOW_TRACKING_URI", "")
+        self.mlflow_uri = mlflow_uri
+        run_id = run or os.environ.get("MLFLOW_RUN_ID")
+        self.run_id = run_id
+        if run_id is not None:
+            print("Found existing run id}")
+            self._start_run(mlflow_run_id=run_id)
+        logging.info(f"MLFlow logger info: experiment name: {experiment_name}, run id: {run_id} URI: {mlflow_uri}")
+        if not is_running_in_azure_ml:
+            self._start_run()
+
+    @rank_zero_only
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+        is_epoch_metric = "epoch" in metrics
+        for key, value in metrics.items():
+            # Log all epoch-level metrics without the step information
+            # All step-level metrics with step
+            mlflow.log_metric(key, value, step=None if is_epoch_metric else step)
+
+    @rank_zero_only
+    def log_hyperparams(self, params: Union[argparse.Namespace, Dict[str, Any]]) -> None:
```

Comment:
I copied this from AzureMLLogger so I don't know the exact use case in mind here. Logically I don't think you would call `log_hyperparams` with params = None but maybe there is an edge case where it can occur and break logging?